mobile/lib/modules/home/ui/asset_grid/immich_asset_grid_view.dart

```diff
@@ -326,6 +328,15 @@ class ImmichAssetGridViewState extends State<ImmichAssetGridView> {
 
     final date = widget.renderList.elements[pos % maxLength].date;
 
+    // add haptic feedback only when scrolling through the asset list
+    if (_scrolling &&
```

Comment:
scrolling is the fast drag scrolling using the scroll handle. Is this indented?

For large collections, drag scrolling will vibrate constantly because the months change all the time.

Either disable for drag scrolling (thus only active when slowly scrolling to a new month) or add some throttle (e.g. no more than every 500ms etc.)

Reply:
Yes, this is intended. The haptic feedback "notifies" the user when a new month has been selected.

Debouncing the haptic feedback would only create confusion on the user side.

Reply:
Have you tested the situation (quickly scrolling through a large collection where you jump multiple months a second)?
The dart-java communication does not cope well with being trashed with calls and would just not perform important actions anymore - ultimately leading to worse user experience. We had that issue with something else...

Reply:
Yes, I did test this scenario, and the app worked well (on a Pixel 6 Pro).
But for certain (older) phones this might cause problems.

I think _500ms_ debounce is too long. What do you think of _50ms_ or _100ms_?

Reply:
Alright, that's good to hear!

50ms would already be 20 times per second.
I wonder, can you really differentiate / feel the difference between 10 and 20 haptic feedback per second?
I would throttle it to the value that you can feel a difference when you slightly decrease the rate per second.
If this is 20 times per second, go with 50ms.
If this is 10 times per second, go with 100ms.

Reply:
> Just to make sure, I think you need to use throttle - not debounce.

Yeah, I mixed them up, sorry.


The difference between 50ms and 100ms is noticeable but can be ignored.